### PR TITLE
Various improvements, fixes and extensions

### PR DESCRIPTION
**Pull request by Maarc**
This PR bundles a couple of changes I did on my local rule set.

It comprises changes like:
- title normalization (for example to have all hibernate related issues starting with "Hibernate")
- formatting
- level of effort updated (0->1 and 1 -> 3)
- severity update (mandatory -> optional)
- new rules

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

File: rules-reviewed/eap6/jboss-eap5/jboss-eap5-java.windup.xml [hunk 1]
```diff
@@ -52,6 +52,7 @@
                       ]]>
                     </message>
                     <link title="JBoss EAP 6 - Messaging Configuration" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.4/html/Administration_and_Configuration_Guide/sect-Configuration1.html#Configure_the_JMS_Server1"/>
+                    <link title="JBoss EAP 7 - Messaging Configuration" href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0/configuring-messaging/chapter-3-basic-configuration"/>
```
Comment (mareknovotny) [inline rules-reviewed/eap6/jboss-eap5/jboss-eap5-java.windup.xml]: This is not ruleset for EAP 7 so including link for EAP7 is not correct IMO see https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/eap7/eap5/jboss-eap5-7-java.windup.xml#L52

**Comment by mareknovotny** [comment]
I see adding "Hibernate 4/5" into titles a little problematic as the rules are already tagged with proper version of hibernate and the titles using version(s) are harder to maintain in future in comparison to add/change only meta data in rulesets.

File: rules-reviewed/eap6/websphere/websphere-catchall.windup.xml [hunk 2]
```diff
@@ -17,6 +17,34 @@
...
+                            <message>
+                                This is a reference to the IBM DB2 driver type (`{package}.{type}`). It probably does not need to be migrated. However the IBM DB2 driver should be configured properly.
+                            </message>
+                            <link href="https://access.redhat.com/solutions/93693" title="Configure datasource settings in EAP 6" />
+                        </hint>
+                    </perform>
+                </iteration>
+            </perform>
+            <where param="package">
+                <matches pattern="com\.ibm\.db2\.jcc(\..*)?" />
```
Comment (mareknovotny) [inline rules-reviewed/eap6/websphere/websphere-catchall.windup.xml]: whenever we know the package for a rule is fixed, it is better to use it directly and not in parametr as thjat results in better optimization for while scanning in tree hierarchy of packages.
So this should be better as:
`<javaclass references="com.ibm.db2.jcc.{subpackage}{type}" />` or straightforward as `<javaclass references="com.ibm.db2.jcc.{subpackage-with-type}" />`

**Comment by mareknovotny** [comment]
retest

**Comment by mareknovotny** [comment]
This PR misses test customization to the changes committed and not all changes are really wanted to be in the core rules. 

We discussed this PR on our standup mtg and were decided to take changes as https://github.com/windup/windup-rulesets/pull/173, where are some commits skipped as they are not correct for every migration and that PR is fixing/adding tests too.

**Pull request closed without merging**